components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/editor/js/design-view/cron-generator/cron-generator.js

```diff
@@ -0,0 +1,771 @@
+                return false;
+            }
+            var list = month.split(",");
+            return list.every(function (month){
+                if(month.includes('/')) {
+                    var startingDayOfMonthOptionArr = month.split('/');
+                    var isValidElements = (isValidateNumericValue([startingDayOfMonthOptionArr[0]], 1, 12) &&
+                                                        isValidateNumericValue([startingDayOfMonthOptionArr[1]], 1, 12));
+                    var isValidFirstElem = (startingDayOfMonthOptionArr[0] === '*' &&
+                                                        isValidateNumericValue([startingDayOfMonthOptionArr[1]], 1, 12));
+                    return isValidElements || isValidFirstElem;
+
+                } else if(month.includes('-')) {
+                    var monthRangeArr = month.split('-');
+                    var validMonthRange = parseInt(monthRangeArr[0]) < parseInt(monthRangeArr[1]);
+                    var validMonthStrRange = MONTH_LIST.indexOf(monthRangeArr[0]) < MONTH_LIST.indexOf(monthRangeArr[1]);
+
+                    return !isNaN(parseInt(monthRangeArr[0])) && !isNaN(parseInt(monthRangeArr[1])) ?
+                        isValidateNumericValue(monthRangeArr, 1, 12) && validMonthRange :
+                        isValidateStringValue(monthRangeArr, MONTH_LIST) && validMonthStrRange;
+                } else if(typeof month === 'string') {
+                    var firstIndexValue = month.charAt(0);
+                    var secondIndexValue = month.charAt(1);
+                    if(month.length === 1){
+                        return isValidateNumericValue([month], 1, 12)
+                    } else {
+                        return (!isNaN(parseInt(firstIndexValue)) && !isNaN(parseInt(secondIndexValue))) ?
+                                isValidateNumericValue([month], 1, 12) : isValidateStringValue([month], MONTH_LIST);
+                    }
+                } else {
+                    return false;
+                }
+            });
+        };
+
+        var WEEK_ARRRAY = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat','sun'];
+
+        var isValidDayOfWeekValue = function(dayOfWeek, dayOfMonth) {
+            if((dayOfWeek === '*' && dayOfMonth !== '*') || (dayOfWeek === '?' && dayOfMonth !== '?')) {
+                return true;
+            }
+            if(dayOfWeek === 'L') {
+                return true;
+            }
+            if(typeof dayOfWeek === 'string' && dayOfMonth === '?' && dayOfWeek.includes('L')) {
+                if(dayOfWeek.includes(',') && dayOfMonth === '?'){
+                    return false;
+                }
+                if (!isNaN(parseInt(dayOfWeek))){
+                    return isValidateNumericValue([dayOfWeek], 1, 7);
+                }
+            }
+            if(dayOfWeek.includes('_')){
+                return false;
+            }
+            if (dayOfWeek.search(/[^\w-,*\/#]/) !== -1) {
```

Comment:
put constants to reges with meaningful names

Reply:
fixed in 0fb87f3